filebeat/input/filestream/input.go

```diff
@@ -400,3 +451,39 @@ func (inp *filestream) eventFromMessage(m reader.Message, path string) beat.Even
 		Fields:    m.Fields,
 	}
 }
+func (inp *filestream) Monitor(ctx input.Context, src loginp.Source) {
+	if !inp.detailedMetricsEnabled {
+		return
+	}
+
+	log := ctx.Logger.With("state-id", src.Name())
+
+	fs, ok := src.(fileSource)
+	if !ok {
+		log.Error("not a filesource, not monitoring detailed metrics of file")
+		return
+	}
+
+	log = ctx.Logger.With("path", fs.newPath)
+
+	monitorCtx, cancel := ctxtool.WithFunc(ctx.Cancelation, func() {
+		log.Debug("Stopping detailed metrics collection of file")
+	})
+	ctx.Cancelation = monitorCtx
+
+	inp.detailedMetrics[src.Name()] = newProgressMetrics(inp.ID, fs.newPath, src.Name(), cancel)
+	go func() {
+		timed.Periodic(ctx.Cancelation, 30*time.Second, func() error {
+			// os.Stat can be used here because Monitor is only used when
+			// the file itself is not opened, so we do not increase the
+			// number of open files in the Beat.
+			fi, err := os.Stat(fs.newPath)
```

Comment:
true, but even a file that is not opened might be renamed. If we have a logs rotation event, we might suddenly collect the wrong stats for the current source.